### ISSUE TITLE
Dot notation behavior 

### ISSUE THREAD

**Issue by rhysemmerson** [inertiajs/inertia-laravel#355]
After updating the package we had a test failure which highlighted some strange behavior with the dot notation.

When using dot notation in props the original key remains in the array.

Taking this example from the relevant test
```PHP
[
    'auth' => [
        'user' => [
            'name' => 'Jonathan Reinink',
        ],
    ],
    'auth.user.can.deleteProducts' => true,
    'product' => ['name' => 'My example product'],
]
```

The nested prop has been set on the user prop but the original key `auth.user.can.deleteProducts` remains 
```
array:3 [
  "auth" => array:1 [
    "user" => array:2 [
      "name" => "Jonathan Reinink"
      "can" => array:1 [
        "deleteProducts" => true
      ]
    ]
  ]
  "auth.user.can.deleteProducts" => true
  "product" => array:1 [
    "name" => "My example product"
  ]
]
```

The other thing I noticed and what caused the test failure  (this didn't seem to be the case before) was that nested props within props are also nested.
```PHP
[
  'auth' => [
      'user' => [
          'name' => 'Jonathan Reinink',
          'foo.bar' => 'yes',
      ],
  ]
]
```

```
array:1 [
  "auth" => array:1 [
    "user" => array:3 [
      "name" => "Jonathan Reinink"
      "foo.bar" => "yes"
      "foo" => array:1 [
        "bar" => "yes"
      ]
    ]
  ]
]
```

This isn't wrong but it does have implications for at least one of our api resources which returns a nested key with a dot like this

**Comment by rhysemmerson** [comment]
I had a play around with returning different versions of the array but I couldn't get it to leave foo.bar alone. 

My question is does this behavior make sense for keys that exist on a prop or should it only apply to the top level?

**Comment by claudiodekker** [comment]
Hey @rhysemmerson,

Ugh, apologies for all the issues you're running into these last few releases. The original key should indeed disappear, for which I'll add a new test. I'll also make sure that dot notation only works on the top-level.

**Comment by claudiodekker** [comment]
Fixed in 507b0a0ad8321028b8651528099f73a88b158359

**Comment by rhysemmerson** [comment]
@claudiodekker thanks for the fast turn around!
